### PR TITLE
Added lerna

### PR DESCRIPTION
**Pull request by migue1coel6odev**
This PR tries to get [lerna](https://github.com/lerna/lerna) working onto diana - _lerna_ is a tool for managing JavaScript projects with multiple packages.

For this to be possible that was a few required changes: 
- Moved modules from src directory to a modules dir on project root.
- Each package contains now a package.json responsable to have the package dependencies and  version. They also contain an own tsconfig.json that extends from the root. 
- The rollup tool is removed as we won't need a bundler at this point since the app that is going to be using this components will probably have a bundler taking care of it.

TODO:
- add readme's to each package
- add tests
- clean up unused stuff
- scripts to publish/prettier/linting
- correct build:module (adding lib dir)

### CODE REVIEW

File: modules/modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import ReactModal, {
+  Props as ReactModalProps,
+  setAppElement
+} from "react-modal";
+import { withStyles } from "@diana-ui/base";
+import { WithStylesProps, ThemeStyleSheetFactory } from "@diana-ui/types";
+
+export interface IProps extends WithStylesProps, ReactModalProps {
+  title: string;
```
Comment (nunoliveira12) [inline modules/modal/Modal.tsx]: why do we have this props for Modal?

File: modules/modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      }
+    }}
+    overlayClassName={cx(styles.overlay)}
+    {...props}
+  >
+    {children}
+  </ReactModal>
+);
+
+export const ModalSetAppElement = setAppElement;
```
Comment (nunoliveira12) [inline modules/modal/Modal.tsx]: why is this needed?

File: modules/modal/ModalFooter.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { withStyles } from "@diana-ui/base";
+import { ThemeStyleSheetFactory, WithStylesProps } from "@diana-ui/types";
+
+export interface IProps extends WithStylesProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  actions: any[];
```
Comment (nunoliveira12) [inline modules/modal/ModalFooter.tsx]: why actions?

File: modules/modal/ModalHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import { withStyles } from "@diana-ui/base";
+import { ThemeStyleSheetFactory, WithStylesProps } from "@diana-ui/types";
+import { Icon } from "@diana-ui/icon";
+import { SectionTitle, Description } from "@diana-ui/typography";
+
+export interface IProps extends WithStylesProps {
+  title: string;
+  onClose: () => void;
```
Comment (nunoliveira12) [inline modules/modal/ModalHeader.tsx]: onClose should be on Modal. And he would pass it to is children (if Modal header). So that modal onclose would be the same for both. You can close via esc, via click overlay, via click close and would be the same

**Comment by nunoliveira12** [comment]
We leave it open about "withStyles(ModalHeader)". I know it was my plan, but later it might be an issue if we extend ModalHeader with a different name. Anyway, I think it is fine for now and probably almost all cases

File: modules/modal/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    }
+    setIsOpen(false);
+  };
+
+  return (
+    <ReactModal
+      isOpen={isOpen}
+      className={cx(styles.content)}
+      overlayClassName={cx(styles.overlay)}
+      onRequestClose={e => onClose(e)}
```
Comment (acaldas) [inline modules/modal/Modal.tsx]: This should be `onRequestClose={onClose}` to avoid creating a new function in every render. (probably doesn't make a difference)

Comment (migue1coel6odev) [reply modules/modal/Modal.tsx]: you're right, fixed
